Library/Homebrew/exceptions.rb

```diff
@@ -435,11 +435,23 @@ def initialize(flags)
 # the compilers available on the user's system
 class CompilerSelectionError < RuntimeError
   def initialize(formula)
-    super <<-EOS.undent
-      #{formula.full_name} cannot be built with any available compilers.
-      To install this formula, you may need to:
-        brew install gcc
-      EOS
+    if MacOS.version > :tiger
+      super <<-EOS.undent
+        #{formula.full_name} cannot be built with any available compilers.
+        To install this formula, you may need to:
+          brew install gcc
+        EOS
```

Comment:
Could just build the string outside the if, append the lines to the string in the if and then `super` the string? I don't feel strongly though so feel free to punt on it.